### PR TITLE
Bau: bypass connector node

### PR DESCRIPTION
**Pull request by chrisholmes**
First, it introduces the ability to consume metadata from multiple different Nodes defined by
configuration.  This is supported by the introduction of:
* a `MetadataResolverRepository` to keep track of `MetadataResolvers` tied to a Node (via `entityId`)
* a `MetadataBackedCountrySignatureValidator` which allows for
validation of signatures from multiple countries
* primitives for configuration `CountryConfiguration`,
`EidasMetadataConfiguration`, and `TrustStoreConfiguration`.

This change also introduces a CountryRepository which provides a map of
country code to entity id.

This is used after selecting a country to find the correct entity ID
that an AuthnRequest should be sent to. In addition, the retrieved
entity ID is stored in the session token so that it can be retrieved
when we receive a response later.

After receiving the Response the entity ID in the token is now used to
verify it by checking the Issuer against the value.

This means that `EidasAuthnRequestGenerator` and `ResponseHandler` no
longer require a `eidasEntityId` constructor parameter so the related
config has been removed.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -4,15 +4,23 @@ export PORT=50400
...
...9Z+6vjbDenncXTuncN6r4IdY6BxwOv5Iy4NMH8Y+QDjtgRbJt5mVaBqulj6lKEIsdIm4fc8CI6wcHZuxsvkpWy2XJQ+ShlLU8KHhwd1k5HABWk86tsIneNjdbOSeVlLAJjRw2jsLUcRsXrgpxEcmmfrJSAqUmdXDTjo8G0A8OaOs1FYTHt2F6ju1tW49LABz2qNhU39zShIZW0IpWEdjPEDbzg3Em96WUwtEEcqVm5ciHnV7b8LeBWMXDR1qOrv3SxMcKVaPALvmoNnp+mQI0wu0xL4BVBsTWe3YurknBpmt66anR+qmLgeO2jKvG8ojjxqne5UCAwEAAaNgMF4wDgYDVR0PAQH/BAQDAgeAMAwGA1UdEwEB/wQCMAAwHQYDVR0OBBYEFA8rfaXUsyx5/HtjYrviE+6yjT0ZMB8GA1UdIwQYMBaAFLS9hHnhhKMm5+AH6yp/5MBDyU4KMA0GCSqGSIb3DQEBDQUAA4ICAQCEBEizRh6b4AL15YN937/l6jSt0O/4JgBWZgA4JmUFKSnV+jriA5sKXpYyFnDjl6V7zKOjVhM5QRdpCysP3KRc3Onq/vrOaMWmFKzATynUlP4bYf1oUE/66rXDJouUXw0Zn9KzcLb2AIArKq2nwuS5dcT9kk60egBysdPAYhp0t3rindxh/gII6C85Gl8kj2a6cJjRi+ErKrAP4mgdD8KBxTIgWmwnrHsaZAfvHJeTmFL7P6FsXQPIgOGb6JE5GjeVVdv7cuBir4eX0vthdz/jInc9kHbLRbeQrnJ6KfAzBLgqxOWXIXnJyQBbziwQ24eSt8ZWWtTU73JSSD0tgErGaP1Y+EKmM4wlhIOvPs9rRcWmEBGToGCSzb7loPif+X+pVy9JCsWvLELFP9cBIfITWgbc6AM7/yYTOvCIw5BBBn0dDPiZfgkPhuiExzCQ8k26bPRcmM9eugqcmSLxiKT2aCQnr/VVjdZZnTHvF5kKPihLG2x1Nn7c6wxuCIdk51YYyukpE9uT30psrgDLczqKishudJSvpD4PcdsANE1JrElAg3yL6h4yPEHrIb8RkhMerXhSvf3kVF9GM0g5zM4gLY863+Fn2adAotIYA8emb4sjhgIfs9jAkwcsOj2zoro5AXHI2hj6MZBZPWH5kotUN57lnomNy4i/V2EoWx1ee2DvNGSZkXXloO6/QFRF/NQC1dG9
```
Comment (richardTowers) [inline .env]: Could we call this base64Keystore instead? (If that's what it is...)

File: src/main/java/uk/gov/ida/eidas/bridge/configuration/EidasMetadataConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    @JsonProperty
+    private Integer maxRefreshDelay;
+
+    /* Used to set {@link org.opensaml.saml2.metadata.provider.AbstractReloadingMetadataProvider#maxRefreshDelay} */
+    @Valid
+    @NotNull
+    @JsonProperty
+    private JerseyClientConfiguration client;
+
+    /* Used to set {@link org.opensaml.saml2.metadata.provider.AbstractReloadingMetadataProvider#maxRefreshDelay} */
```
Comment (richardTowers) [inline src/main/java/uk/gov/ida/eidas/bridge/configuration/EidasMetadataConfiguration.java]: Comment shouldn't be on this field.

File: .env (hunk 1 above)
Comment (chrisholmes) [inline .env]: yes, I think I have an idea of what to do

Comment (chrisholmes) [inline .env]: I think it might take a bit of though so if it's alright I'll raise it as a different PR

**Pull request merged**